### PR TITLE
Adding conda-store

### PR DESCRIPTION
**Pull request by costrouc**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by costrouc** [comment]
Will be working to simplify the requirements to make the conda-store build on all architectures.

File: recipes/conda-store-split/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{% set name = "conda-store-split" %}
```
Comment (xhochy) [inline recipes/conda-store-split/meta.yaml:1]:
```suggestion
{% set name = "conda-store" %}
```

File: recipes/conda-store-split/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+{% set version = "0.2.5" %}
+{% set build = 0 %}
+
+package:
+  name: {{ name|lower }}
```
Comment (xhochy) [inline recipes/conda-store-split/meta.yaml:6]:
```suggestion
  name: {{ name|lower }}-split
```

File: recipes/conda-store-split/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  summary: "Conda Environment Management, Builds, and Serve"
+  description: |
+    End users think in terms of environments not packages. The core
+    philosophy of conda-store is to serve identical conda environments
+    in as many ways as possible. Conda Store controls the environment
+    lifecycle: management, builds, and serving of environments.
+  doc_url: https://conda-store.readthedocs.io/
+  dev_url: https://github.com/Quansight/conda-store
+
+extra:
```
Comment (xhochy) [inline recipes/conda-store-split/meta.yaml:84]:
```suggestion
extra:
  feedstock-name: conda-store
```

**Review by xhochy — COMMENTED** [review]
Made a small suggestion that will give you a nicer feedstock reponame / README once merged.

**Comment by jaimergp** [comment]
Thanks for the hints, @xhochy!

Looks like it builds now, but we still need to decide how to interpin the subpackages.

File: recipes/conda-store-split/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
+{% set name = "conda-store" %}
+{% set version = "0.2.5" %}
+{% set build = 0 %}
```
Comment (xhochy) [inline recipes/conda-store-split/meta.yaml:3]:
```suggestion
```

File: recipes/conda-store-split/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+source:
+  - url: https://pypi.io/packages/source/c/conda-store/conda_store-{{ version }}.tar.gz
+    sha256: bbfdfd0f8224e128147523edc0d44c400fdeb164e5053454c15b81db306843f2
+    folder: conda-store
+  - url: https://pypi.io/packages/source/c/conda-store-server/conda-store-server-{{ version }}.tar.gz
+    sha256: 4622e5e32ee7948db41ee1e9767bf8fc9bdc260a57665c85fb3c6110db35de48
+    folder: conda-store-server
+
+build:
+  number: {{ build }}
```
Comment (xhochy) [inline recipes/conda-store-split/meta.yaml:18]:
```suggestion
  number: 0
```

File: recipes/conda-store-split/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+outputs:
+  - name: "conda-store"
+    version: {{ version }}
+    build:
+      number: {{ build }}
```
Comment (xhochy) [inline recipes/conda-store-split/meta.yaml:24]:
```suggestion
```

File: recipes/conda-store-split/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        - {{ pin_subpackage('conda-store-server', exact=True) }}
+
+    test:
+      imports:
+        - conda_store
+
+  - name: "conda-store-server"
+    version: {{ version }}
+    build:
+      number: {{ build }}
```
Comment (xhochy) [inline recipes/conda-store-split/meta.yaml:45]:
```suggestion
```

File: recipes/conda-store-split/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        - pyyaml
+        - pydantic
+        - minio
+        - typer
+      run_constrained:
+        - {{ pin_subpackage('conda-store', exact=True) }}
+
+    test:
+      requires:
+        - conda-store
```
Comment (xhochy) [inline recipes/conda-store-split/meta.yaml:69]: This probably won't fail as `run_exports` and `run_constrained` aren't properly applied inside a single recipe.

**Review by xhochy — COMMENTED** [review]
Keep the build number global and simple, otherwise the bot cannot increase it on automatic PRs.

File: recipes/conda-store-split/meta.yaml (hunk 8 above)
Comment (jaimergp) [reply recipes/conda-store-split/meta.yaml:66]: I see. Should I remove it or just leave it?

Comment (jaimergp) [reply recipes/conda-store-split/meta.yaml]: Turns out it results in a dependency conflict. I will debug locally once the noarch artifacts are enabled. #15223

**Comment by jaimergp** [comment]
I've done some local tests and I think it works to our expectations. Ready for review, @conda-forge/help-python! Thanks!

**Pull request merged**